whisperv6/whisper.go

```diff
@@ -407,12 +407,40 @@ func (whisper *Whisper) AllowP2PMessagesFromPeer(peerID []byte) error {
 // which are not supposed to be forwarded any further.
 // The whisper protocol is agnostic of the format and contents of envelope.
 func (whisper *Whisper) RequestHistoricMessages(peerID []byte, envelope *Envelope) error {
+	return whisper.RequestHistoricMessagesWithTTL(peerID, envelope, 0)
+}
+
+func (whisper *Whisper) RequestHistoricMessagesWithTTL(peerID []byte, envelope *Envelope, ttl time.Duration) error {
```

Comment:
`WithTimeout`? `TTL` has a very specific meaning in Whisper so I would avoid using it in a different meaning.